packages/fleather/lib/src/widgets/autoformats.dart

```diff
@@ -38,6 +38,18 @@ class AutoFormats {
     ]);
   }
 
+  /// Create a new instance of [AutoFormats] with the provided [autoFormats] + fallback [AutoFormats].
```

Comment:
Could you please make current auto format implementations (`_AutoTextDirection`, `_MarkdownInlineShortcuts`, `_MarkdownLineShortcuts`) public so that users have the option of choosing from?

Reply:
Ok, Took a second swing here. Removed buildWithFallback and just extended the fallback method. I added a public enum and some params (autoFormats and defaultAutoFormats) so you can control how fallback builds the AutoFormats class. It will gracefully fall back to its original behavior if no params are provided, or if you just add your own params it'll add the fallback params automatically. But now you can also build a list of desired fallback formats and it will allow you to pick and choose as desired.

I left the merge function I originally had in there because it seems ergonomic for future use but its unused at this point so it could be removed.

This is the commit with my changes: https://github.com/fleather-editor/fleather/pull/372/commits/41e72fc255277faa2989d3a81e2be09f00f363cc

Then I realized I had left a spread in there from an earlier rendition of my changes so I made a second commit to just strip that out: https://github.com/fleather-editor/fleather/pull/372/commits/bf1fb43f8a684de7c2b74d8f1e04109a1ecaf4ac

Happy to make further changes as needed.

Reply:
I'm gonna suggest a change for this if you don't mind.

Reply:
Oh.... Thats fine with me. I would have to track down how they became private to begin with. I didn't write that bit. I'll check it out soon.

Reply:
They were made private from the first commit they were added. Removing the underscore in their names would make them public.